CMakeLists.txt

```diff
@@ -111,18 +111,16 @@ find_package(PkgConfig)
 # In CMake <3.15, the system is searched before the virtual environment.
 if (NOT Python_EXECUTABLE)
     # If the python interpreter isn't specified as a command line option, look for it:
-    if (CMAKE_VERSION VERSION_LESS 3.12)
-      find_package(PythonInterp REQUIRED)
-      set(Python_EXECUTABLE ${PYTHON_EXECUTABLE})
-    else ()
-      find_package(Python COMPONENTS Interpreter REQUIRED)
-    endif ()
+    find_package(Python COMPONENTS Interpreter REQUIRED)
 endif ()
 
 # =================================== MPI
 if (USE_MPI)
   get_property(REQUIRED_MPI_COMPONENTS GLOBAL PROPERTY ENABLED_LANGUAGES)
   list(REMOVE_ITEM REQUIRED_MPI_COMPONENTS CUDA)  # CUDA does not have an MPI component
+  if (NOT CMAKE_CROSSCOMPILING)  # when cross compiling, assume the user know what he/she's doing
```

Comment:
suggestion: use gender-neutral pronoun "they" for the user instead of he/she.
(Reference: https://en.wikipedia.org/wiki/Singular_they)